modules/dynamic/src/smithy4s/dynamic/internals/DynamicModelCompiler.scala

```diff
@@ -210,6 +210,60 @@ private[dynamic] object Compiler {
     override def blobShape(id: ShapeId, shape: BlobShape): Unit =
       update(id, shape.traits, bytes)
 
+    override def enumShape(id: ShapeId, shape: EnumShape): Unit = {
+      val values: List[(Document, EnumValue[Document])] =
+        shape.members.foldMap { m =>
+          m.toList
+            .flatMap { case (k, m) =>
+              getTrait[smithy.api.EnumValue](m.traits).toList.map { v =>
+                (k, v)
+              }
+            }
+            .zipWithIndex
+            .map { case ((k, v), i) =>
+              v.value ->
+                EnumValue(
+                  k,
+                  i,
+                  v.value,
+                  k,
+                  Hints.empty
+                )
+            }
+        }
+
+      val fromDoc = values.toMap
+      val theEnum = enumeration(fromDoc, values.map(_._2))
+      update(id, shape.traits, theEnum)
+    }
+
+    override def intEnumShape(id: ShapeId, shape: IntEnumShape): Unit = {
+      val values: List[EnumValue[Int]] = shape.members.foldMap { m =>
+        m.toList
+          .flatMap { case (k, m) =>
+            getTrait[smithy.api.EnumValue](m.traits).toList.map { v =>
+              (k, v)
+            }
+          }
+          .collect { case (k, smithy.api.EnumValue(Document.DNumber(v))) =>
+            (k, v.toInt) // safe cast?
+          }
+          .map { case (k, v) =>
+            EnumValue(
+              k, // check
```

Comment:
this sounds correct, I guess the `stringName` of int enums is the same as the enum entry's name. In codegen:

```scala
sealed abstract class FaceCard(_value: String, _name: String, _intValue: Int) extends Enumeration.Value {
  override val value: String = _value
  override val name: String = _name
  override val intValue: Int = _intValue
  override val hints: Hints = Hints.empty
  @inline final def widen: FaceCard = this
}
object FaceCard extends Enumeration[FaceCard] with ShapeTag.Companion[FaceCard] {
  val id: ShapeId = ShapeId("smithy4s.example", "FaceCard")

  val hints : Hints = Hints(
    smithy.api.Documentation("FaceCard types"),
    IntEnum(),
  )

  case object JACK extends FaceCard("JACK", "JACK", 1)
  case object QUEEN extends FaceCard("QUEEN", "QUEEN", 2)
  case object KING extends FaceCard("KING", "KING", 3)
  case object ACE extends FaceCard("ACE", "ACE", 4)
  case object JOKER extends FaceCard("JOKER", "JOKER", 5)

  val values: List[FaceCard] = List(
    JACK,
    QUEEN,
    KING,
    ACE,
    JOKER,
  )
  implicit val schema: Schema[FaceCard] = enumeration(values).withId(id).addHints(hints)
}
```